paymentsheet/src/androidTest/java/com/stripe/android/paymentelement/EmbeddedPaymentElementImmediateActionRowSelectionTest.kt

```diff
@@ -0,0 +1,448 @@
+            embeddedContentPage.clickOnSavedPM(card1.id)
+            assertRowSelectionLabel(
+                actualList = rowSelectionLabels,
+                expectedList = listOf(getCardLabel("4242"))
+            )
+
+            testContext.markTestSucceeded()
+        }
+    }
+
+    @Test
+    fun testSavedCardRowReselection_withFormSheetActionContinue() {
+        runEmbeddedPaymentElementRowSelectionTest(
+            responseTestBodyFileName = "elements-sessions-deferred_payment_intent_no_link.json",
+            shouldSetupV1PaymentMethodsResponse = true,
+            configureBlock = {
+                customer(PaymentSheet.CustomerConfiguration("cus_123", "ek_test"))
+                formSheetAction(EmbeddedPaymentElement.FormSheetAction.Continue)
+            }
+        ) { testContext, rowSelectionLabels ->
+            embeddedContentPage.clickOnSavedPM(card1.id)
+            Espresso.onIdle()
+            embeddedContentPage.clickOnSavedPM(card1.id)
+            assertRowSelectionLabel(
+                actualList = rowSelectionLabels,
+                expectedList = listOf(getCardLabel("4242"), getCardLabel("4242"))
+            )
+
+            testContext.markTestSucceeded()
+        }
+    }
+
+    @Test
+    fun testRemoveCard_doesNotInvokeCallback() {
+        runEmbeddedPaymentElementRowSelectionTest(
+            responseTestBodyFileName = "elements-sessions-deferred_payment_intent_no_link.json",
+            shouldSetupV1PaymentMethodsResponse = true,
+            configureBlock = {
+                customer(PaymentSheet.CustomerConfiguration("cus_123", "ek_test"))
+            }
+        ) { testContext, rowSelectionLabels ->
+            embeddedContentPage.clickViewMore()
+            Espresso.onIdle()
+            managePage.waitUntilVisible()
+            managePage.clickEdit()
+            managePage.clickEdit(card1.id)
+            editPage.waitUntilVisible()
+
+            networkRule.setupPaymentMethodDetachResponse(card1.id)
+
+            editPage.clickRemove()
+            Espresso.onIdle()
+            managePage.waitUntilVisible()
+            managePage.waitUntilGone(card1.id)
+            managePage.clickDone()
+
+            assertRowSelectionLabel(rowSelectionLabels, emptyList<String>())
+
+            testContext.markTestSucceeded()
+        }
+    }
+
+    private suspend fun assertRowSelectionLabel(actualList: List<String?>, expectedList: List<String?>) {
+        withContext(Dispatchers.Main) {
+            Espresso.onIdle()
+            assertThat(actualList).isEqualTo(expectedList)
+        }
+    }
+
+    private fun runEmbeddedPaymentElementRowSelectionTest(
+        responseTestBodyFileName: String,
+        shouldSetupV1PaymentMethodsResponse: Boolean = false,
+        configureBlock: EmbeddedPaymentElement.Configuration.Builder.() -> EmbeddedPaymentElement.Configuration.Builder = { this },
+        testBlock: suspend (EmbeddedPaymentElementTestRunnerContext, MutableList<String?>) -> Unit,
```

Comment:
```suggestion
        testBlock: suspend (EmbeddedPaymentElementTestRunnerContext) -> Unit,
```

We typically but anything needed in the test block in the context object, as it scales better when adding future params.